src/aws-elasticache/serverless-cache.ts

```diff
@@ -0,0 +1,500 @@
+import {
+  IResource,
+  Lazy,
+  Names,
+  Resource,
+  Stack,
+  aws_elasticache,
+  aws_ec2,
+  Token,
+  aws_iam,
+  ArnFormat,
+} from 'aws-cdk-lib';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { Construct } from 'constructs';
+import { DailySnapshotTime } from './daily-snapshot-time';
+import { IUserGroup } from './user-group';
+import { Engine } from './util';
+
+/**
+ * A ElastiCache Serverless Cache
+ */
+export interface IServerlessCache extends IResource, aws_ec2.IConnectable {
+  /**
+   * The serverless cache ARN.
+   *
+   * @attribute
+   */
+  readonly serverlessCacheArn: string;
+
+  /**
+   * The serverless cache name.
+   *
+   * @attribute
+   */
+  readonly serverlessCacheName: string;
+
+  /**
+   * The DNS hostname of the cache node.
+   *
+   * @attribute
+   */
+  readonly endpointAddress: string;
+
+  /**
+   * The port number that the cache engine is listening on.
+   *
+   * @attribute
+   */
+  readonly endpointPort: number;
+
+  /**
+   * Grant the given identity the specified actions.
+   */
+  grant(grantee: aws_iam.IGrantable, ...actions: string[]): aws_iam.Grant;
+
+  /**
+   * Grant the given identity connection access to the cache.
+   */
+  grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant;
+}
+
+/**
+ * The version number of the engine the serverless cache is compatible with.
+ */
+export enum MajorVersion {
+  /**
+   * Version 7
+   */
+  VER_7 = '7',
+}
+
+/**
+ * Properties for defining a ElastiCache Serverless Cache.
+ */
+export interface ServerlessCacheProps {
+  /**
+   * The engine the serverless cache is compatible with.
+   */
+  readonly engine: Engine;
```

Comment:
Shouldn't we default it to Valkey? :)

Reply:
Personally, I would also prefer defaul "Valkey"... LoL.
However, I'm not sure if it's appropriate, so I'll wait for the maintainer's opinion.